core/src/com/unciv/ui/EmpireOverviewScreen.kt

```diff
@@ -324,25 +324,41 @@ class EmpireOverviewScreen(val viewingPlayer:CivilizationInfo) : CameraStageBase
     }
 
 
-    fun playerKnows(civ:CivilizationInfo) = civ==viewingPlayer ||
+    private fun playerKnows(civ:CivilizationInfo) = civ==viewingPlayer ||
             viewingPlayer.diplomacy.containsKey(civ.civName)
 
-    fun getDiplomacyGroup(): Group {
+    private fun getDiplomacyGroup(): Group {
         val relevantCivs = viewingPlayer.gameInfo.civilizations.filter { !it.isBarbarian() && !it.isCityState() }
         val freeWidth = stage.width
         val freeHeight = stage.height - topTable.height
-        val groupSize = if (freeWidth > freeHeight) freeHeight else freeWidth
         val group = Group()
-        group.setSize(groupSize,groupSize)
+        group.setSize(freeWidth,freeHeight)
         val civGroups = HashMap<String, Actor>()
+        val civLines = HashMap<String, MutableSet<Actor>>()
         for(i in 0..relevantCivs.lastIndex){
             val civ = relevantCivs[i]
 
             val civGroup = getCivGroup(civ, "", viewingPlayer)
 
             val vector = HexMath.getVectorForAngle(2 * Math.PI.toFloat() *i / relevantCivs.size)
             civGroup.center(group)
-            civGroup.moveBy(vector.x*groupSize/3, vector.y*groupSize/3)
+            civGroup.moveBy(vector.x*freeWidth/3, vector.y*freeHeight/3)
+            civGroup.onClick {
+
+                val numberOfDisabled = civLines.values.count { it.firstOrNull()?.isVisible == false }
+                if (numberOfDisabled == 0)
+                    // all enabled: disable all except current one
+                    civLines.forEach{ (s, set) -> set.forEach {it.isVisible = s == civ.civName} }
+                else {
+                    val chosenOn = civLines[civ.civName]?.firstOrNull()?.isVisible
+                    if (chosenOn != false && numberOfDisabled >= civLines.values.size-1)
```

Comment:
civLines.values.size-1 looks like magic, why not civLines.size? 
Also, why not chosenOn==true? What happens when chosenOn is null?
I confuse =/

Reply:
You are asking the right questions. =)
There is `chosenOn != false` for the cases when user clicks on "dead" or "unknown" civilization which has no relations on the map. In such cases `civLines[civ.civName]` can be `null`, but it still counts as "disabled". So both for the situations when we click on the only enabled civilization (size-1), and when all civilizations are disabled (numberOfDisabled == size) there is `numberOfDisabled >= civLines.values.size-1`.

Reply:
I think a simpler solution would be to just return from the onClick immediately if the civ has nothing in civLines =)
That way chosenOn can be a 'real' boolean value, and a lot of the code will be simpler =D

Reply:
I will need to rethink the concept. Will come back to this later. Might be, there is even more elegant solution.

Reply:
I have re-written the  handler of visibility, so now it performs better, and I hope the code is more understandable.